pkg/component/component_test.go

```diff
@@ -272,3 +149,115 @@ func getUnstructured(name, kind, apiVersion, managed, componentType, namespace s
+	resourceDeploy2.SetLabels(labels.Builder().WithMode(labels.ComponentDeployMode).Labels())
+
+	otherResource := unstructured.Unstructured{}
+
+	packageManifestResource := unstructured.Unstructured{}
+	packageManifestResource.SetKind("PackageManifest")
+	packageManifestResource.SetLabels(labels.Builder().WithMode(labels.ComponentDevMode).Labels())
+
+	type args struct {
+		client    func(ctrl *gomock.Controller) kclient.ClientInterface
+		name      string
+		namespace string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []api.RunningMode
+		wantErr bool
+	}{
+		{
+			name: "No resources",
+			args: args{
+				client: func(ctrl *gomock.Controller) kclient.ClientInterface {
+					c := kclient.NewMockClientInterface(ctrl)
+					c.EXPECT().GetAllResourcesFromSelector(gomock.Any(), gomock.Any()).Return([]unstructured.Unstructured{packageManifestResource, otherResource}, nil)
+					return c
+				},
+				name:      "aname",
+				namespace: "anamespace",
+			},
+			want: []api.RunningMode{},
+		},
+		{
+			name: "Only Dev resources",
+			args: args{
+				client: func(ctrl *gomock.Controller) kclient.ClientInterface {
+					c := kclient.NewMockClientInterface(ctrl)
+					c.EXPECT().GetAllResourcesFromSelector(gomock.Any(), gomock.Any()).Return([]unstructured.Unstructured{packageManifestResource, otherResource, resourceDev1, resourceDev2}, nil)
+					return c
+				},
+				name:      "aname",
+				namespace: "anamespace",
+			},
+			want: []api.RunningMode{api.RunningModeDev},
+		},
+		{
+			name: "Only Deploy resources",
+			args: args{
+				client: func(ctrl *gomock.Controller) kclient.ClientInterface {
+					c := kclient.NewMockClientInterface(ctrl)
+					c.EXPECT().GetAllResourcesFromSelector(gomock.Any(), gomock.Any()).Return([]unstructured.Unstructured{packageManifestResource, otherResource, resourceDeploy1, resourceDeploy2}, nil)
+					return c
+				},
+				name:      "aname",
+				namespace: "anamespace",
+			},
+			want: []api.RunningMode{api.RunningModeDeploy},
+		},
+		{
+			name: "Dev and Deploy resources",
+			args: args{
+				client: func(ctrl *gomock.Controller) kclient.ClientInterface {
+					c := kclient.NewMockClientInterface(ctrl)
+					c.EXPECT().GetAllResourcesFromSelector(gomock.Any(), gomock.Any()).Return([]unstructured.Unstructured{packageManifestResource, otherResource, resourceDev1, resourceDev2, resourceDeploy1, resourceDeploy2}, nil)
+					return c
+				},
+				name:      "aname",
+				namespace: "anamespace",
+			},
+			want: []api.RunningMode{api.RunningModeDev, api.RunningModeDeploy},
+		},
+		{
+			name: "Unknown",
+			args: args{
+				client: func(ctrl *gomock.Controller) kclient.ClientInterface {
+					c := kclient.NewMockClientInterface(ctrl)
+					c.EXPECT().GetAllResourcesFromSelector(gomock.Any(), gomock.Any()).Return(nil, errors.New("error"))
+					return c
+				},
+				name:      "aname",
+				namespace: "anamespace",
+			},
+			want: []api.RunningMode{api.RunningModeUnknown},
+		},
```

Comment:
Could we add a case testing an empty list returned by `GetAllResourcesFromSelector`.
At first sight, I thought this was covered by the `No resources` test, but this makes `GetAllResourcesFromSelector` return resources that are ignored.

Reply:
Yes! I think that made me fix a bug for some edge case
Thanks

Reply:
Indeed. I see the edge case fixed by the filtered list :-D